File: PTA/B_level/1004_成绩排名.py
# -*- coding:UTF-8 -*-
# PTA 1004 成绩排名
print("Please input a integer:")
num = int( input() )
students = []
tags = ('Name', 'Id', 'Score')

while num > 0:
    tmp = {} # init
    info = input()
    items = info.split(" ")
    tmp = { tag:item  for tag, item in zip(tags, items) }
    #print(tmp)
    score = int( tmp[tags[-1]] )
    tmp[tags[-1]] = score # int
    students.append( tmp )
    num -= 1
    
# get all the score
grades = [ student[ tags[-1] ]  for student in students ]
#print("grades", grades)
grades.sort()
#print("grades", grades)

low = grades[0]
high = grades[-1]
grades = [ high, low ] # rellocate
#print("grades", grades)

for i in range( 2 ):
    for d in students:
        #  find the specifc score of correspond student
        if d[ tags[-1] ] == grades[i]:
            print( d[ tags[0] ], d[ tags[1] ], end='' )
            if i == 0:
                print("")    
    

File: PTA/B_level/1004_成绩排名.cpp
#include<iostream>
#include<string>
using namespace std;
struct Student
{
	string name, sno;
	int  score;
};
int main(void)
{
	int n, max1 = -1, min1 = 101, maxnum, minnum;
	Student s[1000];
	cin >> n;
	for ( int i = 0; i <n; i++ )
	{
		cin >> s[i].name >> s[i].sno >> s[i].score;
		if ( s[i].score > max1 )
		{
			max1 = s[i].score;
			maxnum = i;
		}
		if ( s[i].score < min1 )
		{ 
			min1 = s[i].score;
			minnum = i;
		}
	}
	cout << s[maxnum].name << " " << s[maxnum].sno << endl;
	cout<< s[minnum].name << " " << s[minnum].sno;
	return 0;
}
